appserver/payara-appserver-modules/microprofile/opentracing/src/main/java/fish/payara/microprofile/opentracing/jaxrs/JerseyOpenTracingAutoDiscoverable.java

```diff
@@ -58,7 +57,7 @@
     @Override
     public void configure(FeatureContext context) {
         // Only register for application deployments (not the admin console)
-        final ServiceLocator serviceLocator = new PayaraTracingServices().getBasicServiceLocator();
+        final ServiceLocator serviceLocator = Globals.getStaticBaseServiceLocator();
```

Comment:
This is wrong - autodiscoverable is the trap causing problems, Jersey will find it even without Payara runing and try to use it. So safe should be:
final PayaraTracingServices services = new PayaraTracingServices();
```
Deployment deployment = services.getDeployment(); // as other services
if (deployment == null || deployment.getCurrentDeploymentContext() == null) {
  return;
}
...
```
note: in fact the deployment will be a singleton instance of ApplicationLifecycle, says my Eclipse ...

This is about architecture and responsibilities - PayaraTracingServices is a border class, where on one side is Jersey and on the other is HK2. The border covers if it will provide those services or not, and Jersey then has no other dependency on HK2 services. This class has control and responsibility over it.